### PR TITLE
Initial implementation. ...

### PR DESCRIPTION
**Pull request by taphu**
See commits for comments.

### CODE REVIEW

File: canteven-snap.cabal [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    base,
+    blaze-html >= 0.8 && < 0.9,
+    bytestring,
+    data-default,
+    hslogger,
+    markdown >= 0.1 && < 0.2,
+    snap >= 0.14 && < 0.15,
+    template-haskell,
+    text,
+    transformers
```
Comment (potomak) [inline canteven-snap.cabal]: Maybe you could specify an upper and lower bound to these dependencies.

File: canteven-snap.cabal [hunk 2]
```diff
@@ -0,0 +1,38 @@
+-- Initial canteven-snap.cabal generated by cabal init.  For further 
+-- documentation, see http://haskell.org/cabal/users-guide/
+
+name:                canteven-snap
+version:             0.1.0.0
+synopsis:            A more or less random collection of Snap shorthand and utilities that make webservice programming easier.
+-- description:         
+homepage:            https://github.com/taphu/canteven-snap
```
Comment (potomak) [inline canteven-snap.cabal]: @SumAll or @taphu ?

Comment (taphu) [inline canteven-snap.cabal]: sumall.

Comment (taphu) [inline canteven-snap.cabal]: Let me fix this.

**Comment by glasserc** [comment]
Aw crap, I should have gotten to this earlier. Is it possible for us to make these return MonadSnap instances instead of Snap? That makes it easier to wrap them when you're using a monad transformer around your Snap instance.

**Comment by taphu** [comment]
There's nothing stopping you.

**Comment by glasserc** [comment]
Yes yes, fine, sorry, when I said "us", I meant "not me". But sure, PRs
welcome, I got it.

**Pull request merged**